pkg/docker/util.go

```diff
@@ -54,27 +64,88 @@ func GetImageRegistryAuth(auths *client.AuthConfigurations, imageName string) cl
 		glog.V(5).Infof("Using %s credentials for pulling %s", auth.Email, imageName)
 		return auth
 	}
-	return client.AuthConfiguration{}
+	return api.AuthConfig{}
 }
 
 // LoadImageRegistryAuth loads and returns the set of client auth objects from a docker config
 // json file.
-func LoadImageRegistryAuth(dockerCfg io.Reader) *client.AuthConfigurations {
-	auths, err := client.NewAuthConfigurations(dockerCfg)
+func LoadImageRegistryAuth(dockerCfg io.Reader) *AuthConfigurations {
+	auths, err := NewAuthConfigurations(dockerCfg)
 	if err != nil {
 		glog.V(0).Infof("error: Unable to load docker config")
 		return nil
 	}
 	return auths
 }
 
+// begin borrowed from fsouza
```

Comment:
The go-dockerclient method is public, and the json/object schema from
docker of course is.

But continuing to depend on go-dockerclient is as worrisome a prospect, no?

On Thu, Jul 28, 2016 at 2:13 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/docker/util.go
> <https://github.com/openshift/source-to-image/pull/554#discussion_r72672922>
> :
>
> >  	if err != nil {
> >  		glog.V(0).Infof("error: Unable to load docker config")
> >  		return nil
> >  	}
> >  	return auths
> >  }
> >
> > +// begin borrowed from fsouza
>
> And the code they use to hydrate it isn't public? The reason i'm worried
> is that docker has previously changed the format of the config.json file
> and broken us, so i'd rather depend on an api for reading this file than
> doing it ourselves.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/source-to-image/pull/554/files/3de0d0b3b36d99748e143c5118a2921e728b766a..41d1fc0a3fccc90faeb93db9e60c4f701e1dbd4b#r72672922>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADbadD3-mdXx8RT8ewmH0ryuTLlBVdm1ks5qaPEwgaJpZM4JUVnn>
> .
>